game_event/tests.py

```diff
@@ -252,3 +273,48 @@ def test_fail_add_player_to_event_already_in_it(self, client, game_event, player
         assert response.status_code == 200
         with pytest.raises(IntegrityError):
             response = client.post(url, {'answer': 'yes'})
+
+
+@pytest.mark.django_db
+class TestGetWeatherFromApi:
+    def test_one_valid_response(self, game_event, player, mock_urlopen):
+        mock_response = mock.Mock()
+        mock_response.status = 200
+        mock_response.read.return_value = bytes('{"weather": [{"icon": "04d"}],"main": {"temp": 13.19}}', 'utf-8')
+        mock_urlopen.return_value = mock_response
+        output = get_weather_from_api([game_event], mock_urlopen)
+        game_event.weather = {'icon': "http://openweathermap.org/img/wn/04d@2x.png", 'temp': 13.19}
+        assert output == [game_event]
```

Comment:
I think you might be actually comparing the same thing here. The function is mutating the game event object it takes as input and returns it. Then you're using another reference of the same object and compare between the two.

To confirm, you can drop the line of assigning to `game_event.weather` and see if the assertion still passes. If it does, you need to compare against some other object.

Reply:
Hey @yftacherzog you were right we we do compare the same thing.
What other object we can compare to, the problem is that the function we are testing returns a list of game events, so we need to compare a list with another list, but the problem is that we cannot create another game_event object with the same id. wdyt how can we fix it ?

Reply:
In that case, you can fall back to asserting on the weather directly rather than the complete list. We tried testing more thoroughly. Didn't work.